File: w01/d04/INSTRUCTORS/calculator/calculator_spec.rb

require_relative '../wdispec.rb'  # The version you were mananging
# require_relative '../wdispec_instructor.rb'  # The version we refactored together


require_relative 'calculator.rb'

calc = Calculator.new

test = "calculator is an instance of the Calculator class"
actual = calc.class
expected = Object.const_get('Calculator')
assert_equal(actual, expected, test)

test = "calculator has an .add method that adds two numbers"
actual  = calc.add(2,4)
expected = 6
assert_equal(actual, expected, test)


test = "calculator has a .subtract method that subtracts two numbers"
actual = calc.subtract(4,2)
expected = 2
assert_equal(actual, expected, test)

test = "calculator has a .divide method that works"
actual = calc.divide(8,5)
expected = 1.6
assert_equal(actual, expected, test)

test = "calculator has a .multiply method"
actual = calc.multiply(13,13)
expected = 169
assert_equal(actual, expected, test)

test = "it doesn't allow divison by zero"
actual = calc.divide(1,0)
expected = "NaN"
assert_equal(actual, expected)

File: w07/d05/jasmine_drill/spec/calculatorSpec.js
describe('The calculator', function(){

  beforeEach(function(){
    calculator = new Calculator();
    num1 = 2;
    num2 = 3;
  });

  it('correctly adds two numbers', function(){
    var actual = calculator.add(num1, num2);
    var expected = 5;
    expect(actual).toEqual(expected);
  });

  it('correctly subtracts two numbers', function(){
    var actual = calculator.subtract(num1, num2);
    var expected = -1;
    expect(actual).toEqual(expected);
  });

  it('correctly multiplies two numbers', function(){
    var actual = calculator.multiply(num1, num2);
    var expected = 6;
    expect(actual).toEqual(expected);
  });

  it('correctly divides two numbers', function(){
    var num1 = 6;
    var num2 = 2;
    var actual = calculator.divide(num1, num2);
    var expected = 3;
    expect(actual).toEqual(expected);
  });

});
